### PR TITLE
partial implementation of #949 and #930 enhancement adjustments

### PR DESCRIPTION
**Pull request by bluerthanever**
1. removed some codes from implementation full sized cover image enhancement
2. partial implementation of #949, to store urls found in content that are not images, and calls `PixivHelper.write_url_in_description` to write urls to file.
3. updated test cases.

### CODE REVIEW

**Comment by bluerthanever** [comment]
i changed some codes from 930 implementation because i think some are not so necessary, for example the new field for `FanboxPost` which in codes is actually set the same as `coverImageUrl`, or you could say the other way around...
also before the implementation, cover images would be added to embedded files, but you seem to have removed that. so i added that back...

File: PixivModelFanbox.py [hunk 1]
```diff
@@ -159,14 +162,11 @@ def __str__(self):
...
-        self.coverImageUrl = jsPost["coverImageUrl"]
+        coverUrl = jsPost["coverImageUrl"]
         # Issue #930
-        if self.coverImageUrl is not None:
-            original_url = jsPost["coverImageUrl"]
-            self.fullCoverSizeUrl = _re_fanbox_cover.sub("fanbox", original_url)
-            self.coverImageUrl = self.fullCoverSizeUrl
-            # self.try_add(self.fullCoverSizeUrl, self.images)
-            # self.try_add(self.fullCoverSizeUrl, self.embeddedFiles)
+        if not self.coverImageUrl and coverUrl:
```
Comment (Nandaka) [inline PixivModelFanbox.py:167]: If the self.coverImageUrl already set, then it won't set the self.fullCoverSizeUrl anymore.

File: PixivModelFanbox.py [hunk 2]
```diff
@@ -434,11 +450,7 @@ def WriteHtml(self, html_template, useAbsolutePaths, filename):
 
         cover_image = ""
         if self.coverImageUrl:
-            # Issue #930
-            if self.fullCoverSizeUrl:
-                cover_image = f'<div class="cover"><a href="{self.fullCoverSizeUrl}" target="_blank"><img style="max-width: 1200px; height:auto;" src="{self.coverImageUrl}"/></a></div>'
-            else:
-                cover_image = f'<div class="cover"><img src="{self.coverImageUrl}"/></div>'
+            cover_image = f'<div class="cover"><img src="{self.coverImageUrl}"/></div>'
```
Comment (Nandaka) [inline PixivModelFanbox.py:453]: need to adjust the cover size, else it will be too big in the html. That is why I add inline style here.

Comment (bluerthanever) [reply PixivModelFanbox.py:453]: i tested with default html template, and `%coverImage%` is placed under a `<div class="main">` which is under a `<div class="root">`, and in css there are these parts: `.root{max-width:1280px; margin:auto; background-color:#ffffff;}` and `*{margin:0px; padding:0px; max-width:100%; overflow:auto;}`, so the max-width is set to 1280px....
unless the user uses a custom template, in which case the user could modify the style by setting the styles of class "cover" for this `div` and tokens under it.
putting the max-width in the token made it impossible to change it without changing the codes, so i removed it.

also the `href` inside `a` token and the `src` inside `img` token are actually set the same, so i just removed the `a` token wrapping outside the `img` token.

that's my opinion though.

File: PixivModelFanbox.py (hunk 1 above)
Comment (bluerthanever) [reply PixivModelFanbox.py:167]: i removed the `fullCoverSizeUrl` field, and set the `coverImageUrl` to the full sized image url inside the if. =)

File: PixivModelFanbox.py (hunk 2 above)
Comment (Nandaka) [reply PixivModelFanbox.py:453]: gotcha

File: PixivModelFanbox.py (hunk 1 above)
Comment (Nandaka) [reply PixivModelFanbox.py:167]: ok, looks like `coverImageUrl` only set here, so there is no risk that it is already have some values and the big cover is not resolved

File: PixivModelFanbox.py (hunk 2 above)
Comment (bluerthanever) [reply PixivModelFanbox.py:453]: glad we are on the same page
;P

**Pull request merged**